lightning/src/ln/channelmanager.rs

```diff
@@ -10458,20 +10471,28 @@ where
 	fn create_blinded_payment_paths(
 		&self, amount_msats: u64, payment_secret: PaymentSecret, payment_context: PaymentContext
 	) -> Result<Vec<BlindedPaymentPath>, ()> {
+		let expanded_key = &self.inbound_payment_key;
+		let entropy = &*self.entropy_source;
 		let secp_ctx = &self.secp_ctx;
 
 		let first_hops = self.list_usable_channels();
 		let payee_node_id = self.get_our_node_id();
 		let max_cltv_expiry = self.best_block.read().unwrap().height + CLTV_FAR_FAR_AWAY
 			+ LATENCY_GRACE_PERIOD_BLOCKS;
-		let payee_tlvs = ReceiveTlvs {
+
+		let mut payee_tlvs = ReceiveTlvs {
 			payment_secret,
 			payment_constraints: PaymentConstraints {
 				max_cltv_expiry,
 				htlc_minimum_msat: 1,
 			},
 			payment_context,
+			authentication: None,
 		};
+		let nonce = Nonce::from_entropy_source(entropy);
+		let hmac = payee_tlvs.hmac_for_offer_payment(nonce, expanded_key);
+		payee_tlvs.authentication = Some((hmac, nonce));
```

Comment:
I'll need to document how this should be set. But I'm wondering if we should pass the `NodeSigner` to `BlindedPaymentPath::new`, which already has an `EntropySource`. Then anyone creating a `BlindedPaymentPath` wouldn't need to do this explicitly. Drawback is we'd either need to pass both of those to `Router::create_blinded_payment_paths` or initialize `DefaultRouter` with a `NodeSigner` -- it already has an `EntropySource`. 

@TheBlueMatt Not sure if that is really preferable. Thoughts?

Reply:
I was wondering if we could consider passing `expanded_key` as an input parameter to `Router::create_blinded_payment_paths`. This could potentially enable `DefaultRouter` to handle both `HMAC` and `nonce` creation directly within `create_blinded_payment_paths`, setting them appropriately in the process.

However, I noticed that the codebase doesn't seem to have any `pub` functions that take `expanded_key` as an input parameter, so this approach might not be feasible.

Curious to hear your thoughts on this!

Reply:
Ok, I took some inspiration from this and a discussion based on @shaavan's [comment](https://github.com/lightningdevkit/rust-lightning/pull/3435#pullrequestreview-2488795505) to make an `UnauthenticatedReceiveTlvs` with a method to convert to `ReceiveTlvs`. This way we can never have `ReceiveTlvs` in an invalid state since the `Option` is no longer used.